### PR TITLE
[DRAFT] Support custom GRU

### PR DESCRIPTION
**Pull request by BalyshevArtem**
This draft supports custom GRU, adds conversion in circle2cirlce as new pass, and adds new kernel in onert-micro

ONE-DCO-1.0-Signed-off-by: Artem Balyshev <a.balyshev@samsung.com>

### CODE REVIEW

**Comment by chunseoklee** [comment]
@BalyshevArtem Can we extend this to multiple GRU layer pattern like While-while-while-strideslice ?

**Comment by BalyshevArtem** [comment]
> Can we extend this to multiple GRU layer pattern like While-while-while-strideslice ?
I think yes
Could you please provide tensorflow or tflite model to obtain such graph?

**Comment by chunseoklee** [comment]
@BalyshevArtem Let's try to merge this PR into master. For this, we need to do : 

- [ ] extend to multiple GRU layer pattern like While-while-while-strideslice (skip if it can be merged w/o this extension)
- [ ] value check for Custom GRU op

File: compiler/luci/lang/include/luci/IR/Module.h [hunk 1]
```diff
@@ -46,6 +46,8 @@ class Module final
 public:
   void add(std::unique_ptr<loco::Graph> &&g);
 
+  void remove(size_t idx);
```
Comment (seanshpark) [inline compiler/luci/lang/include/luci/IR/Module.h:49]: Is it possible to make a copy of module without this `idx` subgraph and replace it?

File: compiler/luci/pass/include/luci/CircleOptimizer.h [hunk 2]
```diff
@@ -34,6 +34,7 @@ class CircleOptimizer final
   {
     enum Algorithm
     {
+      FuseUnrolledGRUAsCustomGRU,
```
Comment (seanshpark) [inline compiler/luci/pass/include/luci/CircleOptimizer.h:37]: plz add to bottom not on the top

Comment (BalyshevArtem) [reply compiler/luci/pass/include/luci/CircleOptimizer.h:37]: Actually, it is first draft, I will post new one with changes only in `compiler` part.

**Pull request closed without merging**